iOS/FVM/FVM/Highlighting/HighlightHandler.swift

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Flying Vehicle Monster team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SceneKit
+
+internal class HighlightHandler {
+    private let materialProcessor = MaterialProcessor()
+    private let animationDuration: CFTimeInterval = 0.5
+    
+    internal func setHighlightOn(node: SCNNode) {
+        runAnimatedAction {
+            materialProcessor.highlightNewMaterial(forNode: node)
+        }
+    }
+    
+    internal func setHighlightOff(nodeName: String) {
+        runAnimatedAction {
+            materialProcessor.restoreMaterial(for: nodeName)
+        }
+    }
+    
+    internal func setAllHighlightsOff() {
+        runAnimatedAction {
+            materialProcessor.restoreAll()
+        }
+    }
+    
+    internal func isHighlighted(nodeName: String) -> Bool {
+        return materialProcessor.hasKeyFor(nodeName)
+    }
+    
+    private func runAnimatedAction(action: () -> Void) {
+        SCNTransaction.begin()
+        SCNTransaction.animationDuration = animationDuration
+        action()
```

Comment:
I don't like this name. As for me, here lacks a verb or information that is closure in the name.
Maybe acionClosure()? @SzymonSmykala @ghawliczek

Reply:
Isn't run a verb? Why do you need information that there's a closure when you're given parameter type whenever you use this method? XCode even uses trailing closure syntax by default

Reply:
I thought about line 46...
Name 'action' for me isn't clear and I would want to have an information that is closure in this place

Reply:
Well, you've got open-close brackets, meaning that this variable holds something that can be called. Action is pretty usual name, actionClosure sounds odd to me. Kinda like 'peopleCountInt' or 'salaryDouble'

Reply:
Can't agree. I would stick to convention, variables - nouns, methods - verbs